practices/understanding_git.md

```diff
@@ -0,0 +1,105 @@
+It was first created by [Linus Torvalds](https://en.wikipedia.org/wiki/Linus_Torvalds) in 2005 for the development of [Linux kernel](https://en.wikipedia.org/wiki/Linux_kernel).
+
+
+## How Does It Work?
+
+![Git_understanding](/files/Git_i_dont_know.png)</br>
+Nvie: [Git understranding](https://medium.com/hackernoon/https-medium-com-zspajich-understanding-git-data-model-95eb16cc99f5)
+
+Git is lot more complicated than it looks at it's front end command line. Let's have a look at it.
+
+#### Data Model
+
+When a repository is created and some changes are made and committed. Git generates a 40-Character checksum (SHA1) hash. These hashes are created for every object and the first two characters of the checksum are for the directory name and the rest of them are for the file name. 
+There are three types of objects created by git. First, is **blob** object which stores the snapshot of the files that we committed (there can n objects for n committed files). Second, it creates **tree** object which contains the list of all committed files having a pointer to blob objects. The third and final object is the **commit** object that has the pointer to the tree object.
+
+#### Branching
+
+It is an independent line of development for a particular project having its commits. So how does git does the branching? For a layman, it looks like a list of commits but it is not exactly how it works. Git only keeps the track of the latest commit on a branch that can easily be used to reconstruct the whole commit list. It then used a pointer named **head** which has reference to the checkout branch and it is also used for tracking.
+
+#### Index
+
+According to the git's point of view, it has mainly three areas: **Working directory, staging area, and the repository**. When we work on a project on our local machine, it is stored in your computer's filesystem which can be called **working directory**. The changes that are made to the project will remain in the working directory unless we add them to the staging area using the `git add` command. The staging area can be described as the initial state of the new commit. The staging area can be used to fine-tune the commits by adding or removing until it suits you, after which you can do `git commit` to save it to the directory as an object. For this process, git doesn't use any directory but on the other hand, it uses **index** which can be used to track changes.
+
+
+
+## Why Git Is Preferred
+
+- Git is much faster than Subversion.
+- Git's repositories are much smaller than Subversions.
+- Git was designed to be fully distributed from the start, allowing each developer to have full local control.
+- Git branches are simpler and less resource heavy than Subversion's.
+- Git branches carry their entire history.
+- Git provides better auditing of branch and merge events
+- Git's repo file formats are simple, so repair is easy and corruption is rare.
+- Git repository clones act as full repository backups.
+
```

Comment:
Please, look at [Problems covered in DXKB](https://developerexperience.io/problems). Add here a short part about Problems Git Helps to Solve. For example, https://developerexperience.io/problems/poor-code-quality seems as a good fit. If there are other problems which you think Git solves, add them as well. (For example, the Lack of cooperation in a team or the Lack of source control. Don't feel forced to use them, they are just suggestions for you.

Reply:
Ok, I will look into it

Reply:
Please, add Problems Git Helps to solve. 

If a beginner comes to Git as Practice, they have an interest in what issues Git solves.

Reply:
Can you share some, as I looked into the article that you gave here as link and added 2 of them. If would be very helpful if you can guide me with the same. 
Thanks.
like what exact problem are you looking to solve.

Reply:
It is a structural thing :) Problems Git Helps to Solve should have its own "paragraph" and its headline. There add the "problems Git solves" (I have added some ideas, you have some ideas in your text, I think together they are more than enough)

Reply:
So, I should add these points as well.